### ISSUE TITLE
DropdownMenu still cutting off in some cases

### ISSUE THREAD

**Issue by Thijsiez** [JetBrains/compose-jb#1428]
This is a continuation of [JetBrains/compose-jb#1388](https://github.com/JetBrains/compose-jb/issues/1388), where I found a related new issue in `1.0.0-beta6-dev474`. Thank you so much for the quick updates and support so far!
When a `DropdownMenu` is quite close to the bottom of the window, it looks like part of the first item is cut off when opening above the button which opens the menu. This seems to only happen when there's a specific amount of space below the button. In my reproducer below, the changeover point is 96dp.

Box below at 96dp, which behaves as expected:
![96dp space below](https://user-images.githubusercontent.com/2866734/142708815-8efdfed0-a57a-4751-8187-06d130c35894.PNG)

Box below at 95dp, which now cuts the top item:
![95dp space below](https://user-images.githubusercontent.com/2866734/142708829-a1606d67-3ac7-4e5b-8fc0-5e58f0d7c6ea.PNG)

Reproducer:
```
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.material.Button
import androidx.compose.material.DropdownMenu
import androidx.compose.material.DropdownMenuItem
import androidx.compose.material.Text
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.DpSize
import androidx.compose.ui.unit.dp
import androidx.compose.ui.window.Window
import androidx.compose.ui.window.application
import androidx.compose.ui.window.rememberWindowState

fun main() = application {
    val windowState = rememberWindowState(
        size = DpSize(480.dp, 320.dp)
    )
    var expanded by remember { mutableStateOf(false) }
    Window(
        onCloseRequest = ::exitApplication,
        state = windowState,
        title = "Dropdown Test"
    ) {
        Column {
            Spacer(Modifier.weight(1f))
            Box {
                Button(
                    onClick = { expanded = true }
                ) {
                    Text("Open the dropdown")
                }
                DropdownMenu(
                    expanded = expanded,
                    onDismissRequest = { expanded = false }
                ) {
                    DropdownMenuItem(onClick = {}) { Text("First item") }
                    DropdownMenuItem(onClick = {}) { Text("Second item") }
                    DropdownMenuItem(onClick = {}) { Text("Third item") }
                    DropdownMenuItem(onClick = {}) { Text("Fourth item") }
                    DropdownMenuItem(onClick = {}) { Text("Fifth item") }
                    DropdownMenuItem(onClick = {}) { Text("Sixth item") }
                    DropdownMenuItem(onClick = {}) { Text("Seventh item") }
                    DropdownMenuItem(onClick = {}) { Text("Eighth item") }
                }
            }
            Spacer(Modifier.height(96.dp))
        }
    }
}
```

**Comment by BreimerR** [comment]
Can't find component DropdownMenu in compose-jb version 1.0.1 not sure why. Would appreciate some direction
I've implemented compose.material in gradle common

**Comment by akurasov** [comment]
@BreimerR do you want to say that some code works fine with 1.0.0 and doesn't work with 1.0.1? Could you share small reproducer, please?

**Comment by BreimerR** [comment]
@akurasov Just realized DropdownMenu isn't in common but in separate desktop and android implementations just did an expect works fine now 
For multiplatform dropdown
https://github.com/atsushieno/compose-mpp.git

**Comment by akurasov** [comment]
So there is no issue anymore, right?

**Comment by Thijsiez** [comment]
@BreimerR what do you mean you can't find it? If you copy-paste the above code, that'll work with no problems. DropdownMenu is at `androidx.compose.material.DropdownMenu`, always has been 🤔
@akurasov looks like the issue has been fixed somewhere between `1.0.0-beta6-dev474` and `1.0.1`, as the above reproducer works and behaves fine now. I'll close this ticket :)

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5183">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
